### PR TITLE
test(sanity): migrate sanity package to vitest

### PR DESCRIPTION
**Pull request by binoy14**
### Description

Migrates `sanity` tests to vitest

### What to review

Changes makes sense and all tests pass

### Testing

All tests should pass

### Notes for release

N/A

### CODE REVIEW

**Comment by binoy14** [comment]
* **#7573** <a href="https://app.graphite.dev/github/pr/sanity-io/sanity/7573?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7579** <a href="https://app.graphite.dev/github/pr/sanity-io/sanity/7579?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7578** <a href="https://app.graphite.dev/github/pr/sanity-io/sanity/7578?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `next`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @binoy14 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by ryanbonial** [comment]
@binoy14 The test pass, but I am seeing the addition of about 35 warnings about `Warning: An update to ... inside a test was not wrapped in act(...).` warnings.

CleanShot 2024-10-03 at 13 58 22@2x

**Comment by binoy14** [comment]
> @binoy14 The test pass, but I am seeing the addition of about 35 warnings about `Warning: An update to ... inside a test was not wrapped in act(...).` warnings.
> 
> CleanShot 2024-10-03 at 13 58 22@2x

Pretty sure they existed during jest as well but were getting swallowed by jest. Do you see any instances where it could be causing this? 

My suggestion is if other changes looks good we merge this and do a follow up PR that fixes the warnings

**Review by ryanbonial — APPROVED** [review]
🚀  Sounds like a good plan for a follow-up PR that fixes the warnings

**Comment by bjoerge** [comment]
I just rebased this against next and migrated a couple of new tests that came in while this was waiting to be merged.

**Review by bjoerge — DISMISSED** [review]
Thanks for taking this over the finishing line, @binoy14! <3

File: packages/sanity/test/form/index.ts [hunk 1]
```diff
@@ -1,4 +1,3 @@
-export * from './renderArrayOfObjectsInput'
```
Comment (RitaDias) [inline packages/sanity/test/form/index.ts:1]: Just to double check, this is intentional correct?

Comment (RitaDias) [reply packages/sanity/test/form/index.ts:1]: I'm just trying to understand why or perhaps it was implemented somewhere else?

Comment (bjoerge) [reply packages/sanity/test/form/index.ts]: Good catch. Now I'm not so sure. It was not in use, so it's safe, but not sure if either me or Binoy intended to delete it. Added it back – we can always delete it again later.

**Review by RitaDias — APPROVED** [review]
This is great work thank you @binoy14 and @bjoerge you're legends ❤️ ✨

**Pull request merged**